### PR TITLE
Update cloudscale.ch terraform module to `v3.1.1`

### PR DESCRIPTION
**Pull request by bastjan**
Two breaking changes:
- https://github.com/appuio/terraform-openshift4-cloudscale/pull/33
- https://github.com/appuio/terraform-openshift4-cloudscale/pull/29

This PR updates the module and includes a how-to for the required Terraform state migration.

## Checklist

- [x] Keep pull requests small so they can be easily reviewed.
- [x] Update the documentation.
- [x] Categorize the PR by setting a good title and adding one of the labels:
      `bug`, `enhancement`, `documentation`, `change`, `breaking`, `dependency`
      as they show up in the changelog

### CODE REVIEW

**Comment by simu** [comment]
What's the plan re https://github.com/appuio/terraform-openshift4-cloudscale/pull/29 now? Didn't we discuss including that change in the component upgrade?

**Comment by bastjan** [comment]
> What's the plan re [appuio/terraform-openshift4-cloudscale#29](https://github.com/appuio/terraform-openshift4-cloudscale/pull/29) now? Didn't we discuss including that change in the component upgrade?

Yes. I missed setting the MR to draft.

File: docs/modules/ROOT/pages/how-tos/upgrade-cloudscale-v2-v3.adoc [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+----
++
+.Verify state using `plan`
+[source,bash]
+----
+# No resources, except the hieradata git checkout, should be recreated.
+terraform plan
+----
+
+NOTE: Terraform may wants to create the hieradata git checkout. This is expected and can be ignored.
```
Comment (simu) [inline docs/modules/ROOT/pages/how-tos/upgrade-cloudscale-v2-v3.adoc:125]:
```suggestion
NOTE: Terraform may want to create the hieradata git checkout. This is expected and can be ignored.
```

File: docs/modules/ROOT/pages/how-tos/upgrade-cloudscale-v2-v3.adoc [hunk 2]
```diff
@@ -0,0 +1,125 @@
+= Upgrade cloudscale.ch Terraform module from v2 to v3
+
+https://github.com/appuio/terraform-openshift4-cloudscale/releases/tag/v3.0.0[terraform-openshift4-cloudscale v3.0.0] moves load balancer state to a shared module.
```
Comment (simu) [inline docs/modules/ROOT/pages/how-tos/upgrade-cloudscale-v2-v3.adoc:3]:
```suggestion
When migrating from v2 to v3, the following breaking changes require manual changes to the Terraform state of existing clusters:

* https://github.com/appuio/terraform-openshift4-cloudscale/releases/tag/v3.0.0[terraform-openshift4-cloudscale v3.0.0] uses the new https://github.com/appuio/terraform-modules/tree/main/modules/vshn-lbaas-cloudscale[shared load-balancer module] to provision the LBs.
* https://github.com/appuio/terraform-openshift4-cloudscale/releases/tag/v3.1.0[terraform-openshift4-cloudscale v3.1.0] adds support for using an existing cloudscale.ch subnet.
```

File: docs/modules/ROOT/pages/how-tos/upgrade-cloudscale-v2-v3.adoc [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+
+That release includes breaking changes which require manual steps to upgrade existing clusters.
```
Comment (simu) [inline docs/modules/ROOT/pages/how-tos/upgrade-cloudscale-v2-v3.adoc:5]:
```suggestion
```

**Review by simu — APPROVED** [review]
LGTM.

I added a suggestion for an improved description of the breaking changes in the upgrade how-to.

**Pull request merged**